### PR TITLE
Add fipscheck package

### PR DESCRIPTION
**Pull request by 0xba1a**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Adding new package - fipscheck

###### Change Log  

- Add new package - fipscheck

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local clean build
- Execution of commands in a VM running the vhdx image

### CODE REVIEW

File: SPECS/LICENSES-AND-NOTICES/data/licenses.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
                 "ebtables",
                 "extra-cmake-modules",
                 "fuse-zip",
+                "fipscheck",
```
Comment (PawelWMS) [inline SPECS/LICENSES-AND-NOTICES/data/licenses.json:40]: Should go before `fuse-zip`.

File: SPECS/fipscheck/fipscheck.spec [hunk 2]
```diff
@@ -0,0 +1,156 @@
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Summary:	A library for integrity verification of FIPS validated modules
```
Comment (PawelWMS) [inline SPECS/fipscheck/fipscheck.spec:3]: Please run the spec linter against any new spec.

File: SPECS/fipscheck/fipscheck.spec [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+%package devel
+Summary:	Development files for %{name}
+
+Requires:	%{name}-lib%{?_isa} = %{version}-%{release}
+
+%description devel
+This package contains development files for %{name}.
+
+%prep
+%setup -q
```
Comment (PawelWMS) [inline SPECS/fipscheck/fipscheck.spec:40]: Ideally use `%autosetup`.

File: SPECS/fipscheck/fipscheck.spec [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+%build
+%configure --disable-static
+
+make %{?_smp_mflags}
```
Comment (PawelWMS) [inline SPECS/fipscheck/fipscheck.spec:45]: Try using `%make_build` and other `make` macros whenever possible. This way we always include Mariner's build flags and configuration.

File: SPECS/fipscheck/fipscheck.spec [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+# Add generation of HMAC checksums of the final stripped binaries
+%define __spec_install_post \
+    %{?__debug_package:%{__debug_install_post}} \
+    %{__arch_install_post} \
+    %{__os_install_post} \
+    $RPM_BUILD_ROOT%{_bindir}/fipshmac -d $RPM_BUILD_ROOT%{_libdir}/fipscheck $RPM_BUILD_ROOT%{_bindir}/fipscheck $RPM_BUILD_ROOT%{_libdir}/libfipscheck.so.1.2.1 \
```
Comment (PawelWMS) [inline SPECS/fipscheck/fipscheck.spec:52]: One thing to watch out for when running the spec linter (unless that was fixed while I was on vacation): the linter may attempt to replace `%{_libdir}` and its 64 bit counterpart with `%{_lib}`, which is not the same thing. We need to stick to the `*dir` ones.

File: SPECS/fipscheck/fipscheck.spec [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+find $RPM_BUILD_ROOT -type f -name "*.la" -delete
+
+mkdir -p $RPM_BUILD_ROOT%{_libdir}/fipscheck
+
+%post -p /sbin/ldconfig
+
+%postun -p /sbin/ldconfig
+
+%files
+%{!?_licensedir:%global license %%doc}
```
Comment (PawelWMS) [inline SPECS/fipscheck/fipscheck.spec:70]: Lince can be removed - `_licensedir` is always defined for Mariner.

File: SPECS/fipscheck/fipscheck.spec [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+%{_libdir}/fipscheck/libfipscheck.so.*.hmac
+
+%files devel
+%{_includedir}/fipscheck.h
+%{_libdir}/libfipscheck.so
+%{_mandir}/man3/*
+
+%changelog
+* Tue Aug 03 2021 Bala <balakumaran.kannan@microsoft.com> 1.5.0-9
+- Initial CBL-Mariner import from Fedora (license: BSD)
```
[the remark this answers is not in the document]
Comment (PawelWMS) [reply SPECS/fipscheck/fipscheck.spec:90]: Thanks for catching that, Falak!

**Review by PawelWMS — CHANGES_REQUESTED** [review]
A few small comments to fix, otherwise looks good. Please double-check the GitHub checks are green (except for the linter check, which can sometimes be ignored).

File: SPECS/fipscheck/fipscheck.spec (hunk 5 above)
Comment (0xba1a) [reply SPECS/fipscheck/fipscheck.spec:52]: Seems it is fixed. linter didn't replace `%{_libdir}` with `%{_lib}`.

**Pull request merged**